samza-core/src/main/java/org/apache/samza/operators/spec/InputOperatorSpec.java

```diff
@@ -33,11 +33,16 @@
  */
 public class InputOperatorSpec<K, V> extends OperatorSpec<KV<K, V>, Object> { // Object == KV<K, V> | V
 
-  private final StreamSpec streamSpec;
-  private final Serde<K> keySerde;
-  private final Serde<V> valueSerde;
   private final boolean isKeyed;
 
+  /**
+   * The following {@link Serde}s are serialized by the ExecutionPlanner when generating the configs for a stream, and deserialized
```

Comment:
"The following fields ..." (includes streamSpec which is not a Serde)

Also, we don't need to serialize the input stream information in the StreamSpec for this operator?